### PR TITLE
[#11572] Notification feature - admin page delete notification function 

### PR DESCRIPTION
**Pull request by fsgmhoward**
Part of #11572

Demo:
Demo of deletion feature for admin page

Note that, as the entry will be removed from the list, there will be a gap between the index numbers if middle entries are deleted. Upon refresh, new index numbers will be assigned.

### CODE REVIEW

**Review by NicolasCwy — APPROVED** [review]
LGTM, though I thought it would be better if the numbering on the notification is updated on delete. Understand that this can be a point of confusion for the users so is intended.

**Comment by damithc** [comment]
> LGTM, though I thought it would be better if the numbering on the notification is updated on delete. Understand that this can be a point of confusion for the users so is intended.

I'm actually thinking that we don't need these numbers at all. We don't use such indexing anywhere else in TEAMMATES, right?

**Comment by fsgmhoward** [comment]
@damithc I created these index numbers for visual purposes only and also give admin a way to sort things in creation order. If I am to remove this column, would it be necessary to add one more column "creation date" to serve the same purpose?

**Review by ziqing26 — APPROVED** [review]
LGTM!

**Comment by damithc** [comment]
> @damithc I created these index numbers for visual purposes only and also give admin a way to sort things in creation order. If I am to remove this column, would it be necessary to add one more column "creation time" to serve the same purpose?

@fsgmhoward If the creation time is already in the database, yes, go ahead and use that as a column (instead of the index)

**Comment by fsgmhoward** [comment]
> > @damithc I created these index numbers for visual purposes only and also give admin a way to sort things in creation order. If I am to remove this column, would it be necessary to add one more column "creation time" to serve the same purpose?
> 
> @fsgmhoward If the creation time is already in the database, yes, go ahead and use that as a column (instead of the index)

Noted. Will update that in the next commit.

File: src/web/app/pages-admin/admin-notifications-page/admin-notifications-page.component.ts [hunk 1]
```diff
@@ -199,6 +182,26 @@ export class AdminNotificationsPageComponent implements OnInit {
...
 
+  /**
+   * Deletes a notification.
+   */
+  deleteNotificationHandler(notificationId: string): void {
+    this.notificationService.deleteNotification(notificationId)
+      .subscribe(
+        () => {
+          this.statusMessageService.showSuccessToast('Notification deleted successfully.');
+          this.notificationsTableRowModels = this.notificationsTableRowModels.filter(
```
Comment (jianhandev) [inline src/web/app/pages-admin/admin-notifications-page/admin-notifications-page.component.ts:193]:
```suggestion
        () => {
          this.statusMessageService.showSuccessToast(resp.message);
          this.notificationsTableRowModels = this.notificationsTableRowModels.filter(
```
Not wrong, but can simply use the message from the response.

**Review by jianhandev — APPROVED** [review]
LGTM

File: src/web/app/pages-admin/admin-notifications-page/admin-notifications-page.component.ts (hunk 1 above)
Comment (fsgmhoward) [reply src/web/app/pages-admin/admin-notifications-page/admin-notifications-page.component.ts]: Thank you very much! I have updated that in the latest commit.

**Pull request merged**